### DISCUSSION TITLE
how to keep file names same with previous build

### DISCUSSION THREAD

**Discussion by peguine** [facebook/docusaurus#6927] [Q&A]
When I modify something, and rebuild,  many file names change.

It is a disaster for me to deploy to server, because the root dir has many other files which are not build by docusaurus. So I ]have to delete carefully these old files generated by docusaurus,  then copy new files in, instead of replacing files.

so is there a proper solution?

**Comment by slorber** [comment]
@peguine I don't know what is your deployment process but I suggest that you "merge" your other files with "website/build" on EVERY deploy as part of your deployment process, instead of updating a previous deployment by overriding files.

If you start from an empty folder on every deploy (which you should rather do), then you don't have such problems.

I strongly suggest to use a modern host like Netlify/Vercel with an automated deployment procedure on Git push, instead of uploading files on a FTP or whatever manual deployment procedure you are currently using

---

That being said, all the files that have a hash (`-dfh74k87` suffix) should be located in `website/build/assets` so you can eventually delete these files before adding the docusaurus files of the new build. 

Now if you delete a doc at `docs/myDoc.md`, the file at `website/build/docs/myDoc/index.html` would still exist and there's no easy way for you to automatically delete it, despite the file not changing its name. So I suggest using an automated and modern deployment procedure in all cases to avoid these issues.

**Reply by peguine** [reply]
My website is hosted by apache, and I am using FTP. so how to deploy automatically to apache or nginx?

**Reply by slorber** [reply]
When deploying, you should entirely remove the old files, and provide totally new files, instead to overriding/merging on top of the existing deployment.

Your deployment process is quite outdated: just signup to Vercel and Netlify and get something 100% better that automatically deploys on git push

This is a bit out of the scope of Docusaurus: our only responsibility here is to build the static files to the `/build` folder. Now if you want to use Apache (which we don't recommend) it's your burden to do this right
